awx/main/scheduler/task_manager.py

```diff
@@ -510,9 +486,41 @@ def process_pending_tasks(self, pending_tasks):
                 self.start_task(task, None, task.get_jobs_fail_chain(), None)
                 continue
 
+            # Determine if ther is control capacity for the task
+            control_instance = InstanceGroup.fit_task_to_most_remaining_capacity_instance(
+                task, self.graph['controlplane']['instances'], impact=settings.AWX_CONTROL_NODE_TASK_IMPACT, capacity_type='control'
+            ) or InstanceGroup.find_largest_idle_instance(self.graph['controlplane']['instances'], capacity_type='control')
+            if not control_instance:
+                self.task_needs_capacity(task, tasks_to_update_job_explanation)
+                logger.debug(f"Skipping task {task.log_format} in pending, not enough capacity left on controlplane to control new tasks")
+                continue
+
+            task.controller_node = control_instance.hostname
+
+            # All task.capacity_type == 'control' jobs should run on control plane, no need to loop over instance groups
+            if task.capacity_type == 'control':
+                # This is a task that actually needs to run in the controlplane
+                is_idle = control_instance.jobs_running == 0
+                has_capacity_for_task_impact_and_control = control_instance.remaining_capacity >= (task.task_impact + settings.AWX_CONTROL_NODE_TASK_IMPACT)
+                if not has_capacity_for_task_impact_and_control and not is_idle:
+                    # As in other places, we accept an idle instance if the node with most capacity (the control node we already selected)
+                    # does not have enough capacity.
+                    logger.debug(f"Not enough control capacity on {control_instance} to run {task.log_format}")
+                    self.task_needs_capacity(task, tasks_to_update_job_explanation)
+                    continue
```

Comment:
still want to get this `if` deleted, let me know if you want me to try and contribute a patch or anything.